tests/functional/DatadirTest.php

```diff
@@ -67,5 +74,63 @@ protected function setUp(): void
             ],
         ];
         JsonHelper::writeFile(__DIR__ . '/manual-recursion/expected/data/out/state.json', $state);
+
+        // --- only-new-files tests ---
+        $inputState = [
+            "ex-ftp-state" => [
+                "newest-timestamp" => 0,
+                "last-timestamp-files" => [],
+            ],
+        ];
+        $outputState = [
+            "ex-ftp-state" => [
+                "newest-timestamp" => $timestamps["file_1.txt"],
+                "last-timestamp-files" => ["file_1.txt", "Zvlášť zákeřný učeň s ďolíčky běží podél zóny úlů.csv"],
+            ],
+        ];
+        JsonHelper::writeFile(__DIR__ . '/only-new-files/expected/data/out/state.json', $outputState);
+        JsonHelper::writeFile(__DIR__ . '/only-new-files/source/data/in/state.json', $inputState);
+
+        // -- new-files-from-old-state test --
+        $inputState = [
+            "ex-ftp-state" => [
+                "newest-timestamp" => $timestamps["file_1.txt"],
+                "last-timestamp-files" => ["file_1.txt", "Zvlášť zákeřný učeň s ďolíčky běží podél zóny úlů.csv"],
+            ],
+        ];
+        JsonHelper::writeFile(__DIR__ . '/new-files-from-old-state/source/data/in/state.json', $inputState);
+    }
+
+    /**
+     * @dataProvider provideDatadirSpecifications
+     */
+    public function testDatadir(DatadirTestSpecificationInterface $specification): void
```

Comment:
When we overload `testDatadir()`, it may no longer extends from `DatadirTestCase` but directly from `AbstractDatadirTestCase`

Reply:
I think it's ok, it does the same thing as the method in `DatadirTestCase`, but also creates the file for the specific test.  The other methods will all come from `DatadirTestCase` imho.  nebo nerozumim? :-)

Reply:
Imho stačí `class DatadirTest extends \Keboola\DatadirTests\AbstractDatadirTestCase` místo `\Keboola\DatadirTests\DatadirTestCase`